### ISSUE TITLE
Make GUI data-type aware

### ISSUE THREAD

**Issue by tmr232** [williballenthin/ida-settings#15]
In the current state it is inconvenient to edit anything but numbers.

It would be great if the GUI will display different editing options for different data types. Raw text for strings, numeric input for numbers, a list-view of sorts for lists, a table for dicts...

Further improvements may include editors for "user-defined" types, such as colors.

**Comment by williballenthin** [comment]
I agree this would be a nice improvement over the current interface.

I'm a little suspicious of the potential complexity of determining the types of settings values, while keeping it intuitive to add and change values. Also, handling nested objects would be difficult. I'll happy accept suggestions and patches that implement this feature!
